pkg/api/graph/graph.go

```diff
@@ -67,22 +67,26 @@ type MutableUniqueGraph interface {
 
 type Edge struct {
 	concrete.Edge
-	K string
+	kinds util.StringSet
 }
 
-func NewEdge(head, tail graph.Node, kind string) Edge {
-	return Edge{concrete.Edge{head, tail}, kind}
+func NewEdge(head, tail graph.Node, kinds ...string) Edge {
+	return Edge{concrete.Edge{head, tail}, util.NewStringSet(kinds...)}
 }
 
-func (e Edge) Kind() string {
-	return e.K
+func (e Edge) Kinds() util.StringSet {
+	return e.kinds
+}
+
+func (e Edge) IsKind(kind string) bool {
```

Comment:
I don't like using StringSet if this becomes a very large set, since it's much less efficient than the original check (a single, inline int comparison, vs string equality, vs a map dereference and then a single equality).  It's fine for now.